### PR TITLE
[F2F -879 ] CIC front Test Container Script to run in the pipeline 

### PR DESCRIPTION
**Pull request by mudassar-jafer**
### What changed

- CFN Template  updated to include CIC IPV stub variable
- Added the run local and run tests shell script.
- Fixed github workflows to use test dockerfile.

### Why did it change

Pipeline tests are required to run for CIC codepipeline.

### Issue tracking

- [F2F-879](https://govukverify.atlassian.net/browse/F2F-879)

### Environment variables or secrets

- [ ] Documented in the README
- [ ] Added screenshots to show the implementation is working
- [ ] Ran cfn-lint on any SAM templates

Test report for CIC front tests.

Screenshot 2023-07-21 at 14 58 53
Screenshot 2023-07-21 at 14 58 58

[F2F-879]: https://govukverify.atlassian.net/browse/F2F-879?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: run-tests-locally.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    echo TEST_REPORT_ABSOLUTE_DIR="$TEST_REPORT_DIR" >> docker_vars.env
+    echo TEST_ENVIRONMENT="$ENVIRONMENT" >> docker_vars.env
+    echo ENVIRONMENT="$ENVIRONMENT" >> docker_vars.env
+    echo SAM_STACK_NAME="$SAM_STACK" >> docker_vars.env
+    echo GITHUB_ACTIONS=true >> docker_vars.env
+
+    #clean existing container and image before creating a new one
+    echo "Removing existing containers and images for the test"
+    docker ps -a --filter "ancestor=$DockerImageName" --filter "status=exited" -q | xargs docker rm
+    docker images $DockerImageName -q |xargs docker rmi
```
Comment (darrendunford) [inline run-tests-locally.sh:30]: should we be doing this here if we are using `docker run --rm` ?

File: run-tests.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export GITHUB_ACTIONS=true
+# shellcheck disable=SC2154
+export IPV_STUB_URL=$(remove_quotes $CFN_CICIPVStubExecuteURL)start
+
+cd ./app; npm run test:browser:ci
+error_code=$?
+
+cp -rf /app/test/reports $TEST_REPORT_ABSOLUTE_DIR
+
+exit $error_code
```
Comment (darrendunford) [inline run-tests.sh:22]: do we need to explicitly handle error codes this way if we have used `set -e` at the top of the file?

**Review by darrendunford — DISMISSED** [review]

File: run-tests-locally.sh (hunk 1 above)
Comment (mudassar-jafer) [reply run-tests-locally.sh:30]: This is to make sure we clean up before each run and make sure there are no images left behind form previous runs. For example failed run.

File: yarn.lock
Comment (richa-misra) [inline yarn.lock:1]: yarn.lock seems to have gone to older versions, could it be reverted pls

File: run-tests.sh (hunk 2 above)
Comment (mudassar-jafer) [reply run-tests.sh]: done.

**Review by richa-misra — APPROVED** [review]

**Pull request merged**